libs/partners/anthropic/tests/unit_tests/test_chat_models.py

```diff
@@ -268,3 +272,75 @@ def test_convert_to_anthropic_tool(
     for fn in (pydantic, function, dummy_tool, json_schema, expected, openai_function):
         actual = convert_to_anthropic_tool(fn)  # type: ignore
         assert actual == expected
+
+
+def test__format_messages_with_tool_calls() -> None:
+    system = SystemMessage("fuzz")
+    human = HumanMessage("foo")
+    ai = AIMessage(
+        "",
+        tool_calls=[{"name": "bar", "id": "1", "args": {"baz": "buzz"}}],
+    )
+    tool = ToolMessage(
+        "blurb",
+        tool_call_id="1",
+    )
+    messages = [system, human, ai, tool]
+    expected = (
+        "fuzz",
+        [
+            {"role": "user", "content": "foo"},
+            {
+                "role": "assistant",
+                "content": [
+                    {
+                        "type": "tool_use",
+                        "name": "bar",
+                        "id": "1",
+                        "input": {"baz": "buzz"},
+                    }
+                ],
+            },
+            {
+                "role": "user",
+                "content": [
+                    {"type": "tool_result", "content": "blurb", "tool_use_id": "1"}
+                ],
+            },
+        ],
+    )
+    actual = _format_messages(messages)
+    assert expected == actual
+
+
+def test__format_messages_with_content_and_tool_calls() -> None:
+    system = SystemMessage("fuzz")
+    human = HumanMessage("foo")
+    # If content and tool_calls are specified, content is preferred.
+    ai = AIMessage(
+        "not a tool call",
+        tool_calls=[{"name": "bar", "id": "1", "args": {"baz": "buzz"}}],
+    )
+    tool = ToolMessage(
```

Comment:
```\
    ai = AIMessage(
        [{text content}, {tool call content matching below}, {text content}],
        tool_calls=[{"name": "bar", "id": "1", "args": {"baz": "buzz"}}],
    )
    tool = ToolMessage(
```
should we add a case like this? should theoretically dedupe right?

Reply:
currently ignores tool_calls if content is a list